packages/identity/src/identity/secp256k1.test.ts

```diff
@@ -0,0 +1,117 @@
+import { DerEncodedPublicKey } from '@dfinity/agent';
+import { randomBytes } from 'crypto';
+import { sha256 } from 'js-sha256';
+import Secp256k1 from 'secp256k1';
+import { Secp256k1KeyIdentity, Secp256k1PublicKey } from './secp256k1';
+
+function fromHexString(hexString: string): ArrayBuffer {
+  return new Uint8Array((hexString.match(/.{1,2}/g) ?? []).map(byte => parseInt(byte, 16))).buffer;
+}
+
+// DER KEY SECP256K1 PREFIX = 3056301006072a8648ce3d020106052b8104000a03420004
+const testVectors: Array<[string, string]> = [
+  [
+    '0401ec030acd7d1199f73ae3469329c114944e0693c89502f850bcc6bad397a5956767c79b410c29ac6f587eec84878020fdb54ba002a79b02aa153fe47b6ffd33',
+    '3056301006072a8648ce3d020106052b8104000a0342000401ec030acd7d1199f73ae3469329c114944e0693c89502f850bcc6bad397a5956767c79b410c29ac6f587eec84878020fdb54ba002a79b02aa153fe47b6ffd33',
+  ],
+  [
+    '04eb3cbc81bdfa1d13c410151308b4ac65cc328620e368fe5302e2cf2cb7175132999dc2afff871a68afd66d76dd9c7c3911748e8e54ee2fb71df1a97bb4811c8c',
+    '3056301006072a8648ce3d020106052b8104000a03420004eb3cbc81bdfa1d13c410151308b4ac65cc328620e368fe5302e2cf2cb7175132999dc2afff871a68afd66d76dd9c7c3911748e8e54ee2fb71df1a97bb4811c8c',
+  ],
+  [
+    '04fdac09ea93a1b9b744b5f19f091ada7978ceb2f045875bca8ef9b75fa8061704e76de023c6a23d77a118c5c8d0f5efaf0dbbfcc3702d5590604717f639f6f00d',
+    '3056301006072a8648ce3d020106052b8104000a03420004fdac09ea93a1b9b744b5f19f091ada7978ceb2f045875bca8ef9b75fa8061704e76de023c6a23d77a118c5c8d0f5efaf0dbbfcc3702d5590604717f639f6f00d',
+  ],
+];
+
+test('DER encoding of SECP256K1 keys', async () => {
+  testVectors.forEach(([rawPublicKeyHex, derEncodedPublicKeyHex]) => {
+    const publicKey = Secp256k1PublicKey.fromRaw(fromHexString(rawPublicKeyHex));
+    const expectedDerPublicKey = fromHexString(derEncodedPublicKeyHex);
+    expect(publicKey.toDer()).toEqual(expectedDerPublicKey);
+  });
+});
+
+test('DER decoding of SECP256K1 keys', async () => {
+  testVectors.forEach(([rawPublicKeyHex, derEncodedPublicKeyHex]) => {
+    const derEncoded = Secp256k1PublicKey.fromRaw(fromHexString(rawPublicKeyHex)).toDer();
+
+    const testCase = fromHexString(derEncodedPublicKeyHex);
+
+    expect(derEncoded).toEqual(testCase);
+  });
+});
+
+test('DER decoding of invalid keys', async () => {
```

Comment:
nice that the negative cases are tested!
nit: I'm not a typescript expert, but would it be possible to assert the error message that is thrown?

Reply:
yes, it is possible